extensions/resteasy-reactive/quarkus-resteasy-reactive/runtime/src/main/java/io/quarkus/resteasy/reactive/server/runtime/ResteasyReactiveRecorder.java

```diff
@@ -55,12 +59,49 @@
 @Recorder
 public class ResteasyReactiveRecorder extends ResteasyReactiveCommonRecorder implements EndpointInvokerFactory {
 
+    static final Logger logger = Logger.getLogger("io.quarkus");
+
     public static final Supplier<Executor> EXECUTOR_SUPPLIER = new Supplier<Executor>() {
         @Override
         public Executor get() {
             return ExecutorRecorder.getCurrent();
         }
     };
+    public static final Supplier<Executor> VIRTUAL_EXECUTOR_SUPPLIER = new Supplier<Executor>() {
+        Executor current = null;
+
+        private Executor setVirtualThreadCustomScheduler(Executor executor) throws ClassNotFoundException,
+                InvocationTargetException, InstantiationException, IllegalAccessException, NoSuchMethodException {
+            var vtf = Class.forName("java.lang.ThreadBuilders").getDeclaredClasses()[0];
+            Constructor constructor = vtf.getDeclaredConstructors()[0];
+            constructor.setAccessible(true);
+            ThreadFactory tf = (ThreadFactory) constructor.newInstance(
+                    new Object[] { executor, "quarkus-virtual-factory-", 0, 0,
+                            null });
+
+            return (Executor) Executors.class.getMethod("newThreadPerTaskExecutor", ThreadFactory.class)
+                    .invoke(this, tf);
+        }
+
+        @Override
+        public Executor get() {
+            if (current == null) {
+                try {
+                    current = setVirtualThreadCustomScheduler(
+                            Executors.newFixedThreadPool(Runtime.getRuntime().availableProcessors() * 2));
+                } catch (ClassNotFoundException | InvocationTargetException | InstantiationException
+                        | IllegalAccessException | NoSuchMethodException e) {
+                    //quite ugly but works
+                    logger.warnf("You weren't able to create an executor that spawns virtual threads, the default" +
+                            " blocking executor will be used, please check that your JDK is compatible with " +
+                            "virtual threads");
+                    //if for some reason a class/method can't be loaded or invoked we return the traditional EXECUTOR
+                    current = EXECUTOR_SUPPLIER.get();
+                }
+            }
+            return current;
+        }
+    };
```

Comment:
It would be great to have some comments on why the reflection is needed

Reply:
I added a few lines as comments above the methods, not sure if it does the trick